src/controllers/LiveDataController.ts

```diff
@@ -24,8 +24,14 @@ export async function init() {
     stsApi.onDidLiveProcessDisconnect(resetProcessInfo);
     stsApi.onDidLiveProcessUpdate(updateProcessInfo);
     // proposed API on vscode-spring-boot extension. check before calling.
-    stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);
-    stsApi.onDidLiveProcessMemoryMetricsUpdate?.(updateProcessMemoryMetrics);
+    if(typeof stsApi.onDidLiveProcessGcPausesMetricsUpdate === "function") {
+        stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);
```

Comment:
with above check, here you don't need `?`, which is a null check.

btw `func?.()` roughly equals to `func ? func() : func`

Reply:
I have reverted the check to use `stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);`